nav2_costmap_2d/plugins/obstacle_layer.cpp

```diff
@@ -72,64 +77,61 @@ void ObstacleLayer::onInitialize()
 
   global_frame_ = layered_costmap_->getGlobalFrameID();
   double transform_tolerance;
-  nh.param("transform_tolerance", transform_tolerance, 0.2);
+  node_->get_parameter_or<double>("transform_tolerance", transform_tolerance, 0.2);
 
-  std::string topics_string;
   // get the topics that we'll subscribe to from the parameter server
-  nh.param("observation_sources", topics_string, std::string(""));
-  ROS_INFO("    Subscribed to Topics: %s", topics_string.c_str());
+  std::string topics_string;
+  node_->get_parameter_or<std::string>("observation_sources", topics_string, std::string(""));
+
+  RCLCPP_INFO(node_->get_logger(), "Subscribed to Topics: %s", topics_string.c_str());
 
   // now we need to split the topics based on whitespace which we can use a stringstream for
   std::stringstream ss(topics_string);
 
   std::string source;
   while (ss >> source) {
-    ros::NodeHandle source_node(nh, source);
-
     // get the parameters for the specific topic
     double observation_keep_time, expected_update_rate, min_obstacle_height, max_obstacle_height;
     std::string topic, sensor_frame, data_type;
     bool inf_is_valid, clearing, marking;
-
-    source_node.param("topic", topic, source);
-    source_node.param("sensor_frame", sensor_frame, std::string(""));
-    source_node.param("observation_persistence", observation_keep_time, 0.0);
-    source_node.param("expected_update_rate", expected_update_rate, 0.0);
-    source_node.param("data_type", data_type, std::string("PointCloud"));
-    source_node.param("min_obstacle_height", min_obstacle_height, 0.0);
-    source_node.param("max_obstacle_height", max_obstacle_height, 2.0);
-    source_node.param("inf_is_valid", inf_is_valid, false);
-    source_node.param("clearing", clearing, false);
-    source_node.param("marking", marking, true);
-
-    if (!(data_type == "PointCloud2" || data_type == "PointCloud" || data_type == "LaserScan")) {
-      ROS_FATAL("Only topics that use point clouds or laser scans are currently supported");
+  
+    node_->get_parameter_or<std::string>(source + "." + "topic", topic, source);
+    node_->get_parameter_or<std::string>(source + "." + "sensor_frame", sensor_frame, std::string(""));
+    node_->get_parameter_or<double>(source + "." + "observation_persistence", observation_keep_time, 0.0);
```

Comment:
Do you really need the type specifiers here (`<double>`)? I thought that it was automatically deduced from the variable's type.

Reply:
I can probably remove the specifier